project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -183,6 +183,10 @@ public static boolean shareDocument(String hash, String email) {
     }
 
     long userID = userEntity.getKey().getId();
+    
+    if(getUsersDocumentsHashes(userID).contains(hash)) {
+      return true;
```

Comment:
nitpick: shouldnt we think of if the user already has the hash "shareDocument" should return false? ie the logic is fine, just the naming feels weird to me.

Reply:
I get what you're saying, right now I return false when the result is that the document is not shared when the function is done running. Following this logic, since you are trying to share with a user that is already shared, the end result is we do not share it with them again, but the document is shared with them (from before).